spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin/DefaultZipkinRestTemplateCustomizer.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2013-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.zipkin;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.zip.GZIPOutputStream;
+
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Default {@link ZipkinRestTemplateCustomizer} that can customize interceptor
+ * to the {@link RestTemplate} responsible for sending spans to Zipkin.
+ *
+ * @author Marcin Grzejszczak
+ *
+ * @since 1.1.0
+ */
+public class DefaultZipkinRestTemplateCustomizer implements ZipkinRestTemplateCustomizer {
+	private final ZipkinProperties zipkinProperties;
+
+	public DefaultZipkinRestTemplateCustomizer(
+			ZipkinProperties zipkinProperties) {
+		this.zipkinProperties = zipkinProperties;
+	}
+
+	@Override
+	public void customize(RestTemplate restTemplate) {
+		if (this.zipkinProperties.getCompression().isEnabled()) {
```

Comment:
how is the user supposed to know that the compression property is special?
that it applies unless someone added a customizer? There's no mention of
this caveat unless you read the code.

they're most likely to find out when they notice larger data going across
the network. If you feel strongly that this is a valid separation of
concerns, I'd encourage you to document it, so that the extender knows they
are expected to implement the compression property when customizing.